### PR TITLE
add cart and item model and other related classes

### PR DESCRIPTION
**Pull request by Bohadan-Markatov**

### CODE REVIEW

**Review by AnyaRatusnyak — APPROVED** [review]
Looks good))

File: src/main/java/mate/academy/bookshop/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    private final ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN', 'ROLE_USER')")
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    @Operation(summary = "Add book to cart", description = "Add book to cart")
+    public ShoppingCartResponseDto addBookToCart(
+            Authentication authentication,
+            @RequestBody @Valid CartItemCreateDto cartItemCreateDto
+    ) {
```
Comment (yevheniisynytsia) [inline src/main/java/mate/academy/bookshop/controller/ShoppingCartController.java]:
```suggestion
            @RequestBody @Valid CartItemCreateDto cartItemCreateDto) {

```

File: src/main/java/mate/academy/bookshop/mapper/ShoppingChartMapper.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import mate.academy.bookshop.config.MapperConfig;
+import mate.academy.bookshop.dto.cart.ShoppingCartResponseDto;
+import mate.academy.bookshop.model.ShoppingCart;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class, uses = CartItemMapper.class)
+public interface ShoppingChartMapper {
+    @Mapping(target = "userId", expression = "java(shoppingCart.getUser().getId())")
```
Comment (yevheniisynytsia) [inline src/main/java/mate/academy/bookshop/mapper/ShoppingChartMapper.java]:
```suggestion
    @Mapping(target = "userId", source = "user.id")
```
isn't it working?

File: src/main/java/mate/academy/bookshop/repository/ShoppingCartRepository.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package mate.academy.bookshop.repository;
+
+import mate.academy.bookshop.model.ShoppingCart;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    ShoppingCart findByUserId(Long userId);
```
Comment (yevheniisynytsia) [inline src/main/java/mate/academy/bookshop/repository/ShoppingCartRepository.java]: should be `Opional<ShoppingCart>`

File: src/main/java/mate/academy/bookshop/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Override
+    public ShoppingCartResponseDto getByUserId(Long userId) {
+        return shoppingChartMapper.toDto(shoppingCartRepository.findByUserId(userId));
+    }
+
+    @Override
+    public ShoppingCartResponseDto addBook(Long userId, CartItemCreateDto createDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId);
```
Comment (yevheniisynytsia) [inline src/main/java/mate/academy/bookshop/service/impl/ShoppingCartServiceImpl.java]: what if it is null?

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Well done! Left some minor comments

**Review by yevheniisynytsia — APPROVED** [review]
🔥

**Pull request merged**